rs/boundary_node/rate_limits/api/src/lib.rs

```diff
@@ -11,18 +11,60 @@ pub type RuleId = String;
 pub type IncidentId = String;
 pub type SchemaVersion = u64;
 
-pub type GetConfigResponse = Result<ConfigResponse, String>;
-pub type AddConfigResponse = Result<(), String>;
-pub type GetRuleByIdResponse = Result<OutputRuleMetadata, String>;
+pub type GetConfigResponse = Result<ConfigResponse, GetConfigError>;
+pub type AddConfigResponse = Result<(), AddConfigError>;
+pub type GetRuleByIdResponse = Result<OutputRuleMetadata, GetRuleByIdError>;
 pub type DiscloseRulesResponse = Result<(), DiscloseRulesError>;
-pub type GetRulesByIncidentIdResponse = Result<Vec<OutputRuleMetadata>, String>;
+pub type GetRulesByIncidentIdResponse = Result<Vec<OutputRuleMetadata>, GetRulesByIncidentIdError>;
 
 #[derive(CandidType, Deserialize, Debug, Clone)]
 pub enum DiscloseRulesArg {
     RuleIds(Vec<RuleId>),
     IncidentIds(Vec<IncidentId>),
 }
 
+#[derive(CandidType, Debug, Deserialize)]
+pub enum GetConfigError {
```

Comment:
Do we really need all these errors separately? Maybe just combine in a single enum?

Reply:
`GetRulesByIncidentIdError` and `GetRuleByIdError` could indeed be combined to avoid some overhead, but `GetConfigError` has different variant. My rationale to keep errors separate was for better clarify and for potential forward compatibility. This would ensure we could more easily extend/modify candid errors in the future, if we need to.

Reply:
this was also a recommendation from the `nns` folks - better keep definitions separate and as flexible as possible for the potential future changes